tests/integration/modifiers/did-intersect-test.js

```diff
@@ -3,85 +3,65 @@ import { setupRenderingTest } from 'ember-qunit';
 import sinon from 'sinon';
-
-let intersectionCallback;
-let intersectionOptions;
-let observeStub;
-let disconnectStub;
-let MockIntersectionObserver;
+import Service from '@ember/service';
 
 module('Integration | Modifier | did-intersect', function (hooks) {
   setupRenderingTest(hooks);
 
-  let intersectionObserver;
-
   hooks.beforeEach(function () {
-    intersectionCallback = null;
-    intersectionOptions = null;
-    observeStub = sinon.stub();
-    disconnectStub = sinon.stub();
-
-    MockIntersectionObserver = class MockIntersectionObserver {
-      constructor(callback, options) {
-        intersectionCallback = callback;
-        intersectionOptions = options;
+    class MockObserverManager extends Service {
+      constructor() {
+        super(...arguments);
+        this._admin = {};
       }
 
-      observe = observeStub;
-      disconnect = disconnectStub;
+      observe = sinon.stub();
+      unobserve = sinon.stub();
+      addEnterCallback = sinon.stub();
+      addExitCallback = sinon.stub();
     }
 
-    intersectionObserver = window.IntersectionObserver;
-    window.IntersectionObserver = MockIntersectionObserver;
-
-    this.intersectStub = sinon.stub();
-  });
-
-  hooks.afterEach(function () {
-    window.IntersectionObserver = intersectionObserver;
-  });
-
-  test('modifier integrates with IntersectionObserver', async function (assert) {
-    await render(hbs`<div {{did-intersect this.intersectStub}}></div>`);
+    this.owner.register('service:ember-scroll-modifiers@observer-manager', MockObserverManager);
 
-    assert.ok(intersectionCallback, 'IntersectionObserver received callback');
-    assert.ok(observeStub.calledOnce, 'observe was called');
-    assert.ok(intersectionOptions, 'options passed to IntersectionObserver');
-    assert.equal(typeof intersectionOptions, 'object', 'default options is an object');
-    assert.equal(Object.keys(intersectionOptions).length, 0, 'default options is an empty object');
-
-    let [element] = observeStub.args[0];
-
-    assert.ok(element, 'element was passed to observe');
+    this.observerManagerMock = this.owner.lookup('service:ember-scroll-modifiers@observer-manager');
+    this.enterStub = sinon.stub();
+    this.exitStub = sinon.stub();
   });
 
-  test('modifier triggers handler when IntersectionObserver fires callback', async function (assert) {
-    await render(hbs`<div {{did-intersect this.intersectStub}}></div>`);
+  test('modifier integrates with observer-manager', async function (assert) {
+    assert.expect(3);
 
-    let fakeEntry = { target: {} };
-    let fakeObserver = { observe: {} };
+    await render(hbs`<div {{did-intersect onEnter=this.enterStub onExit=this.exitStub}}></div>`);
 
-    intersectionCallback([fakeEntry], fakeObserver);
-
-    assert.ok(this.intersectStub.calledOnceWith(fakeEntry, fakeObserver), 'handler fired with correct parameters');
+    assert.ok(this.observerManagerMock.observe, 'observerManager received observe call');
```

Comment:
I think for all three of these you want to add `.calledOnce`. All you're asserting right now is that the methods exist.